### ISSUE TITLE
Feedback for “Configuring Workload identity federation between GCP and AWS EKS”

### ISSUE THREAD

**Issue by TheAshwanik** [cloudquery/cloudquery#11688]
Thanks for this article. I see 2 problems from security point of view:

1. You have provided access to all principles under cq-aws-pool, which is not good.
 `gcloud iam service-accounts add-iam-policy-binding \
    cq-fetcher@example-project.iam.gserviceaccount.com \
    --member "principalSet://iam.googleapis.com/projects/123456789012/locations/global/workloadIdentityPools/cq-aws-pool/*" \
    --role "roles/iam.workloadIdentityUser"`

2. This whole setup will use noderole attached to the EKS node, hence any workload sharing the same node will have the same permissions. This again is not good.

**Comment by jsonpr** [comment]
Hi @TheAshwanik, thanks for the feedback!  We've opened a pull request that addresses and clarifies the points you've made here: https://github.com/cloudquery/cloudquery/pull/11904

- For the IAM Policy Binding, that's correct - the `*` provides access to all identities in the pool.  We've added clarification for additional scenarios for policy binding such as using single identities, all identities in a group, or all identities with a specific attribute value along with an example for a single identity.  Keep in mind that using a single identity such as an IAM role may require additional setup including attribute mapping.  For an IAM role, that may look like a custom attribute such as GCP's example `aws_role`.

- For the NodeRole attached to the EKS role, AWS requires creating an IAM role for nodes to use when they are launched.  Before launching nodes and registering them into a cluster, the IAM role for the nodes must be created first.  The [recommendation](https://docs.aws.amazon.com/eks/latest/userguide/create-node-role.html) from AWS is the create an IAM Role with the `AmazonEKSWorkerNodePolicy`, `AmazonEC2ContainerRegistryReadOnly`, and an additional policy or setup depending on networking (Amazon VPC CNI add-on).  We've added clarification to refer to AWS documentation for security best practices as well for EKS.
